plugins/builds/steps/logs.js

```diff
@@ -191,9 +191,19 @@ module.exports = config => ({
                                 maxLines
                             })
                         )
-                        .then(([lines, morePages]) =>
-                            reply(lines).header('X-More-Data', (morePages || !isDone).toString())
-                        );
+                        .then(([lines, morePages]) =>{
+                            if (type !== 'download') {
+                                return reply(lines).header('X-More-Data', (morePages || !isDone).toString());
```

Comment:
we should do this for `lines` in  non-downloads also. UI should not spend time doing any looping.